enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -896,6 +899,81 @@ def can_redeem(self, request, enterprise_customer_uuid):
         )
         return Response(response_serializer.data, status=status.HTTP_200_OK)
 
+    @extend_schema(
+        tags=[SUBSIDY_ACCESS_POLICY_REDEMPTION_API_TAG],
+        summary='Can request.',
+        parameters=[serializers.SubsidyAccessPolicyCanRequestRequestSerializer],
+        responses={
+            status.HTTP_200_OK: serializers.SubsidyAccessPolicyCanRequestElementResponseSerializer(many=True),
+            # TODO: refine these other possible responses:
+            # status.HTTP_403_FORBIDDEN: PermissionDenied,
+            # status.HTTP_404_NOT_FOUND: NotFound,
+        },
+    )
+    @action(
+        detail=False,
+        methods=['get'],
+        url_name='can-request',
+        url_path='enterprise-customer/(?P<enterprise_customer_uuid>[^/.]+)/can-request',
+        pagination_class=None,
+    )
+    def can_request(self, request, enterprise_customer_uuid):
+        """
+        Check if a learner can request access to content. The flow is:
+        1. Find BnR enabled policies first
+        2. Check if content key exists in those policies
+        3. Check for existing pending request by this learner
+        """
+        serializer = serializers.SubsidyAccessPolicyCanRequestRequestSerializer(data=request.query_params)
+        serializer.is_valid(raise_exception=True)
+
+        content_key = serializer.data['content_key']
+        lms_user_id_override = serializer.data.get('lms_user_id') if request.user.is_staff else None
+        lms_user_id = lms_user_id_override or self.lms_user_id or request.user.lms_user_id
```

Comment:
I'm not sure I understood this part, can you explain?

Reply:
It's taken from `can_redeem` endpoint. This allows only staff users to override the `lms_user_id` by passing it as parameter otherwise it picks user's identify from the JWT.